src/v/cluster/controller_backend.cc

```diff
@@ -668,28 +668,31 @@ ss::future<> controller_backend::reconcile_ntp(deltas_t& deltas) {
             ++it;
             continue;
         }
-        auto interrupt_it = find_interrupting_operation(it, deltas);
-        if (interrupt_it != deltas.end()) {
-            vlog(
-              clusterlog.trace,
-              "[{}] cancelling current: {} operation with: {}",
-              it->delta.ntp,
-              *it,
-              *interrupt_it);
-            while (it != interrupt_it) {
-                if (
-                  it->delta.type
-                  == topic_table_delta::op_type::update_properties) {
-                    co_await process_partition_properties_update(
-                      it->delta.ntp, it->delta.new_assignment);
-                }
-                ++it;
-            }
-        }
+
         try {
             auto ec = co_await execute_partition_op(*it);
             if (ec) {
                 if (it->delta.is_reconfiguration_operation()) {
+                    auto interrupt_it = find_interrupting_operation(it, deltas);
```

Comment:
This change looks a bit convoluted to me. The idea of an "interrupting operation" is to determine if the current operation is safe to skip if there is some future operation, right? So if update is now not safe to skip even if followed by cancel, probably that should be reflected directly in `find_interrupting_operation`.

For example, in case of [update, delete] it is still safe to skip to delete, right? But after this patch we will unnecessarily try to execute update.

Reply:
According to our interactive review i change that code to always retry if partition creation failed